include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -298,18 +298,14 @@ struct reduce_over_group
         auto __local_idx = __item_id.get_local_id(0);
         auto __group_size = __item_id.get_local_range().size();
 
-        auto __k = 1;
-
-        do
+        for (_Size __k = 1; __k < __group_size; __k <<= 1)
```

Comment:
We are changing the type of `__k` from auto (with `int` literal) to the same type as `__n`, which is likely a `::std::size_t`.  

The type has to fit the next power of 2 higher than the maximum workgroup size.  Does this mean we can fit in `::std::uint16_t`?  `::std::uint32_t` would be safe certainly.

Reply:
I agree that we could fix `__k` to an unsigned integer. I will change it to `::std::uint32_t` to be safe. There seems to be no performance differences between 16 or 32-bit unsigned integers but using an integer with `auto` seems to be slower.